### PR TITLE
Multiple domains quilting restart

### LINKED ISSUE #1628: Using fv3atm write grid component to write out restart files for multiple domains
## Description
This issue is an follow-up to the issue #1587. In issue #1587, the FV3ATM write grid component is able to write restart files for one domain (either 6 tile global domain or a single regional domain). In this issue, the capability of writing multiple domain restart files will be verified and sample tests will be added.

## Solution
Extend the write grid component restart capability to write out restart files on multiple domains including moving nest domains.

### LINKED ISSUE #1726: Add atm inline post log file
## Description
The write grid component now writes out inline post, history files and restart files. It is requested that when the inline post is finished a log file will be written out so that downstream jobs can be triggered. Similarly a log file will be written out when the write grid component finishes writing history files.  

## Solution
Code will be added to write out the two log files.

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
## Description

This PR updates write grid component to allow writing the restart files for multiple domains (nests)

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1628
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1726

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/268

### CODE REVIEW

File: io/module_wrt_grid_comp.F90 [hunk 1]
```diff
@@ -1003,10 +1005,22 @@ subroutine wrt_initialize_p1(wrt_comp, imp_state_write, exp_state_write, clock,
 !
 !loop over all items in the imp_state_write and count output FieldBundles
 
-      ! if (lprnt) write(0,*)'wrt_initialize_p1: before get_outfile FBCount =', FBCount
+      ! if (lprnt) then
```
Comment (junwang-noaa) [inline io/module_wrt_grid_comp.F90]: Can we remove the prints here and lines 1017-1023?

Comment (DusanJovic-NOAA) [reply io/module_wrt_grid_comp.F90]: Done

**Review by junwang-noaa — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
.gitmodules should be updated if submodules are merged, before my approval.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by jkbk2004** [comment]
@DusanJovic-NOAA https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/268 was merged. can you update hash and revert change in gitmodules?

**Pull request merged**